keanu-project/src/test/java/io/improbable/keanu/tensor/intgr/IntegerScalarTensorShapePreservationTest.java

```diff
@@ -0,0 +1,218 @@
+package io.improbable.keanu.tensor.intgr;
+
+import io.improbable.keanu.tensor.Tensor;
+import org.junit.Before;
+import org.junit.Test;
+import org.nd4j.linalg.api.shape.Shape;
+
+import static org.junit.Assert.assertArrayEquals;
+
+public class IntegerScalarTensorShapePreservationTest {
+
+    private static IntegerTensor intTensor;
+    private static IntegerTensor scalarIntTensor;
+
+    @Before
+    public void setup() {
+        intTensor = IntegerTensor.create(new int[]{5, 6, 7, 8, 9, 10}, new long[]{3, 2});
+        scalarIntTensor = IntegerTensor.create(1, new long[]{1, 1, 1});
+    }
+
+    @Test
+    public void integerScalarMultiplicationPreservesShape() {
+        Tensor multiplicationResult = intTensor.times(scalarIntTensor);
+        resultShapeMatchesBroadcastShape(multiplicationResult, intTensor, scalarIntTensor);
+    }
+
+    @Test
+    public void integerScalarAdditionPreservesShape() {
+        Tensor additionResult = intTensor.plus(scalarIntTensor);
+        resultShapeMatchesBroadcastShape(additionResult, intTensor, scalarIntTensor);
+    }
+
+    @Test
+    public void integerScalarDivisionPreservesShape() {
+        Tensor divisionResult = intTensor.div(scalarIntTensor);
+        resultShapeMatchesBroadcastShape(divisionResult, intTensor, scalarIntTensor);
+    }
+
+    @Test
+    public void integerScalarSubtractionPreservesShape() {
+        Tensor subtractionResult = intTensor.minus(scalarIntTensor);
+        resultShapeMatchesBroadcastShape(subtractionResult, intTensor, scalarIntTensor);
+    }
+
+    @Test
+    public void integerScalarMultiplicationPreservesShapeReverse() {
```

Comment:
nit: put the two Multiplication tests next to each other, to match the pattern in the rest of the file

likewise for Addition, Division, Subtraction

Reply:
done